lib/hal/debug.c

```diff
@@ -27,9 +37,23 @@ static const unsigned char systemFont[] =
 #include "font_unscii_16.h"
 };
 
+static void synchronizeVideoMode()
+{
+	VIDEO_MODE vm = XVideoGetMode();
+	SCREEN_WIDTH = vm.width;
+	SCREEN_HEIGHT = vm.height;
+	SCREEN_BPP = vm.bpp;
+	SCREEN_FB = XVideoGetFB();
```

Comment:
Did you think about adding a `if (SCREEN_FB == NULL) { panic(); }`? Its not necessary the responsibility to check for it, as it can't fail once the user successfully run `XVideoSetMode();`(as they should). But it wouldn't hurt either. 🤔

Reply:
Why would `SCREEN_FB` at `NULL` be an issue? `NULL` is just memory address 0, which is a perfectly valid address in physical memory.

Reply:
oh oh. Yes you are right. I completely forgot that. Sry!